docs/_components/modal.md

```diff
@@ -9,34 +9,36 @@ Modals are a blocking element, they should be used sparingly and only when we ab
 Example:
 
 ```html
-<div class="modal">
+<div class="modal" id="myModal" role="dialog" aria-labelledby="dialog-title" aria-describedby="dialog-description">
     <div class="modal__dialog">
         <div class="modal__content">
             <div class="modal__header">
-                <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
-                <h4 class="modal__title">A simple example</h4>
+                <button type="button" class="close" data-dismiss="modal" aria-label="Close dialog">&times;</button>
+                <h4 class="modal__title" id="dialog-title">A simple example</h4>
             </div>
             <div class="modal__body">
+                <p id="dialog-description" class="sr-only">SR description goes here</p>
                 <p>The modal body might have instructions, a form, or other stuff.</p>
             </div>
             <div class="modal__footer">
-                <button type="button" class="btn" data-dismiss="modal">Cancel</button>
                 <button type="button" class="btn btn--primary">Save Changes</button>
+                <button type="button" class="btn" data-dismiss="modal">Cancel</button>
             </div>
         </div><!-- /.modal__content -->
     </div><!-- /.modal__dialog -->
 </div><!-- /.modal -->
 ```
 
 <div class="pulsar-example">
-    <div class="modal modal__example show">
+    <div class="modal modal__example show" id="myModal-2" role="dialog" aria-labelledby="dialog-title-2" aria-describedby="dialog-description-2">
         <div class="modal__dialog">
             <div class="modal__content">
                 <div class="modal__header">
-                    <button type="button" class="close" data-dismiss="modal" aria-hidden="true">×</button>
-                    <h4 class="modal__title">A simple example</h4>
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close dialog">×</button>
+                    <h4 class="modal__title" id="dialog-title-2">A simple example</h4>
```

Comment:
I feel these examples may be safer to use the main modal title as the basis for internal IDs, so you'd have `modal#myModal2` `h4#myModal2-title` `p#myModal2-description`, which would be easier to explain via documentation and be a bit more obvious when a developer inevitably copies and pastes our code into their UI. 

I suspect the current way would end up with a lot of duplicated ID attributes, we need to be very explicit in the documentation that there are multiple IDs that would need amending if there are multiple modals present within a UI.

It feels quite fragile at the moment... the other option is a new modal helper, but that's out of scope here